api/importer/execute/default_api.go

```diff
@@ -0,0 +1,98 @@
+package execute
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+
+	"github.com/keptn/keptn/api/importer/model"
+)
+
+//go:generate moq -pkg fake --skip-ensure -out ./fake/default_api_mock.go . requestFactory:MockRequestFactory
+
+var /*const*/ ErrTaskFailed = errors.New("task failed")
+
+type otelWrappedHttpClient struct {
+	client http.Client
+}
+
+func (o *otelWrappedHttpClient) Do(r *http.Request) (*http.Response, error) {
+	if _, isOtelTransport := o.client.Transport.(*otelhttp.Transport); !isOtelTransport {
+		o.client.Transport = otelhttp.NewTransport(o.client.Transport)
+	}
+	return o.client.Do(r)
+}
+
+type requestFactory interface {
+	CreateRequest(tCtx model.TaskContext, host string, body io.Reader) (*http.Request, error)
+}
+
+type projectRenderRequestFactory struct {
+	httpMethod string
+	path       string
+}
+
+func (rf *projectRenderRequestFactory) renderUrl(
+	tCtx model.TaskContext, endpoint string, path string,
+) string {
+	return endpoint + strings.Replace(path, "[[project]]", tCtx.Project, -1)
+}
+func (rf *projectRenderRequestFactory) CreateRequest(
+	tCtx model.TaskContext, host string, body io.Reader,
+) (*http.Request,
+	error) {
+	req, err := http.NewRequest(
+		rf.httpMethod,
+		rf.renderUrl(tCtx, host, rf.path),
+		body,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error composing request for api call %s: %w", tCtx.Task.ID, err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	return req, nil
+}
+
+type defaultEndpointHandler struct {
+	requestFactory
+	endpoint string
+}
+
+func (ep *defaultEndpointHandler) ExecuteAPI(doer httpdoer, ate model.APITaskExecution) (any, error) {
+	if ate.Payload != nil {
+		defer ate.Payload.Close()
+	}
+
+	request, err := ep.CreateRequest(ate.Context, ep.endpoint, ate.Payload)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+	response, err := doer.Do(request)
+
+	if err != nil {
+		return nil, fmt.Errorf("error executing request: %w", err)
+	}
+
+	defer response.Body.Close()
+	responseBody := new(any)
+	if response.Header.Get("Content-Type") == "application/json" {
+		bytes, err := io.ReadAll(response.Body)
+		if err == nil {
+			json.Unmarshal(bytes, responseBody)
+		}
+	}
```

Comment:
I'm always expecting JSON as a response from the calls we do. 
However, I'm wondering whether we should log something if it's not JSON, to make troubleshooting easier.

Reply:
Or, alternatively, handle "responseBody" differently in the case of not being application/json?

Reply:
I would add a log in case it's declared json from the content-type but we cannot unmarshal properly...
In case content-type is not jsonm  I am not sure how you would like to represent the body bytes... As a simple string ?

Reply:
@Raffy23 @christian-kreuzberger-dtx  have a look at the latest fixup ;)